tests/components/sensor/test_awair.py

```diff
@@ -0,0 +1,280 @@
+    await setup_awair(hass)
+
+    sensor = hass.states.get('sensor.awair_score')
+    assert sensor.state == '78'
+    assert sensor.attributes['device_class'] == DEVICE_CLASS_SCORE
+    assert sensor.attributes['unit_of_measurement'] == '%'
+
+
+async def test_awair_temp(hass):
+    """Test that we create a temperature sensor."""
+    await setup_awair(hass)
+
+    sensor = hass.states.get('sensor.awair_temperature')
+    assert sensor.state == '22.4'
+    assert sensor.attributes['device_class'] == DEVICE_CLASS_TEMPERATURE
+    assert sensor.attributes['unit_of_measurement'] == TEMP_CELSIUS
+
+
+async def test_awair_humid(hass):
+    """Test that we create a humidity sensor."""
+    await setup_awair(hass)
+
+    sensor = hass.states.get('sensor.awair_humidity')
+    assert sensor.state == '32.73'
+    assert sensor.attributes['device_class'] == DEVICE_CLASS_HUMIDITY
+    assert sensor.attributes['unit_of_measurement'] == '%'
+
+
+async def test_awair_co2(hass):
+    """Test that we create a CO2 sensor."""
+    await setup_awair(hass)
+
+    sensor = hass.states.get('sensor.awair_co2')
+    assert sensor.state == '612'
+    assert sensor.attributes['device_class'] == DEVICE_CLASS_CARBON_DIOXIDE
+    assert sensor.attributes['unit_of_measurement'] == 'ppm'
+
+
+async def test_awair_voc(hass):
+    """Test that we create a CO2 sensor."""
+    await setup_awair(hass)
+
+    sensor = hass.states.get('sensor.awair_voc')
+    assert sensor.state == '1012'
+    assert (sensor.attributes['device_class'] ==
+            DEVICE_CLASS_VOLATILE_ORGANIC_COMPOUNDS)
+    assert sensor.attributes['unit_of_measurement'] == 'ppb'
+
+
+async def test_awair_dust(hass):
+    """Test that we create a pm25 sensor."""
+    await setup_awair(hass)
+
+    # The Awair Gen1 that we mock actually returns 'DUST', but that
+    # is mapped to pm25 internally so that it shows up in Homekit
+    sensor = hass.states.get('sensor.awair_pm25')
+    assert sensor.state == '6.2'
+    assert sensor.attributes['device_class'] == DEVICE_CLASS_PM2_5
+    assert sensor.attributes['unit_of_measurement'] == 'µg/m3'
+
+
+async def test_awair_unsupported_sensors(hass):
+    """Ensure we don't create sensors the stubbed device doesn't support."""
+    await setup_awair(hass)
+
+    # Our tests mock an Awair Gen 1 device, which should never return
+    # PM10 sensor readings. Assert that we didn't create a pm10 sensor,
+    # which could happen if someone were ever to refactor incorrectly.
+    assert hass.states.get('sensor.awair_pm10') is None
+
+
+async def test_availability(hass):
+    """Ensure that we mark the component available/unavailable correctly."""
+    await setup_awair(hass)
+
+    assert hass.states.get('sensor.awair_score').state == '78'
+
+    future = NOW + timedelta(minutes=30)
+    data_patch = patch('python_awair.AwairClient.air_data_latest',
+                       return_value=mock_coro(AIR_DATA_FIXTURE))
+    time_patch = patch('homeassistant.util.utcnow',
```

Comment:
Why are we patching this place? Are we using that path anywhere? I think we should patch `'homeassistant.util.dt.utcnow'`. That will patch the place that event helpers are using.

Reply:
Yes, it is needed. I swear it's not just adding code for the sheer fun of it. :smile: 

In [homeassistant/util/\_\_init.py\_\_](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/util/__init__.py#L17), we import `utcnow` like so:
```python
from .dt import as_local, utcnow
```

I think that python mocking is not as smart as we'd like to assume. I feel like the import path matters somehow - importing it at the base of `util` seems to change the import path. Regardless, it's better to just show, not tell. Here's with and without the patched method in question: https://gist.github.com/ahayworth/16bd4985c0105382d2247374d5e0f25e

Reply:
Ok, I found where we use that path. It's in the `Throttle` class in `homeassistant.util`. So you're right that we need to patch that. But we should also patch the path that I requested above, since that is used during setup component.